### PR TITLE
Feature/86 refactor text field component

### PR DESCRIPTION
**Pull request by thomastepi**
Refactored TextFieldComponents to enhance customization by introducing the CustomTextField component. This change allows each rendered text field to be modified with specific props. Additionally, PropTypes have been added to CustomTextField to enforce prop validation.

**Changes Made:**

1.  Implemented CustomTextField component to replace direct usage of MUI TextField.
2.  Introduced PropTypes to validate props passed to CustomTextField.
3.  Created test cases to verify component functionality.

**Reason for Changes:**

To streamline component reusability and maintain consistency in prop usage throughout the application.

### CODE REVIEW

**Review by erenfn — CHANGES_REQUESTED** [review]
- The default font size should be 13px.
- Also, please adjust the height of the text field accordingly.
- It would be very helpful if you could provide a brief explanation of the prop types as a block comment in the file.
- Some details, such as shadows, were missing. You can find these in the Style Guide Figma file.
- It would be good if you could specify the default parameters.

Other than that, good work!

File: frontend/src/tests/scenes/login/LoginPage.test.jsx [hunk 1]
```diff
@@ -1,41 +1,40 @@
-import { describe, it, expect } from 'vitest';
-import { render, screen, fireEvent } from '@testing-library/react';
-import LoginPage from '../../../scenes/login/LoginPage';
+import { describe, it, expect } from "vitest";
```
Comment (erenfn) [inline frontend/src/tests/scenes/login/LoginPage.test.jsx:1]: Also why do you have these files? I think they are not merged into the develop branch

**Comment by thomastepi** [comment]
> * The default font size should be 13px.
> * Also, please adjust the height of the text field accordingly.
> * It would be very helpful if you could provide a brief explanation of the prop types as a block comment in the file.
> * Some details, such as shadows, were missing. You can find these in the Style Guide Figma file.
> * It would be good if you could specify the default parameters.
> 
> Other than that, good work!

I have implemented the changes you recommended and also updated the file to align with the styling in the Figma guide. thanks

File: frontend/src/components/TextFieldComponents/CustomTextField.jsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        }}
+        FormHelperTextProps={{
+          sx: { margin: 0, paddingTop: 1 },
+        }}
+      />
+    </Box>
+  );
+};
+
+// Explanation of the PropTypes used in this file
```
Comment (swoopertr) [inline frontend/src/components/TextFieldComponents/CustomTextField.jsx:79]: those descriptions should be in readme.md file with an appropriate explanation.

File: frontend/src/components/TextFieldComponents/CustomTextField.jsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  labelFontWeight,
+  TextFieldWidth,
+  inputHeight,
+}) => {
+  return (
+    <Box paddingY={1}>
+      <InputLabel className="label" sx={{ fontWeight: labelFontWeight }}>
+        {labelText}
+      </InputLabel>
+      <TextField
```
Comment (swoopertr) [inline frontend/src/components/TextFieldComponents/CustomTextField.jsx:26]: this component usage is too complex, hard to understand, and difficult to modify. Please split the more basic functions or objects.

File: frontend/src/components/TextFieldComponents/TextFieldComponents.jsx [hunk 4]
```diff
@@ -1,93 +1,111 @@
...
+
+        <CustomTextField
+          labelText="Email"
+          labelFontWeight={600}
+          inputHeight="34px"
+          TextFieldWidth="320px"
           defaultValue="olivia@untitledui.com"
           error
           helperText="This is an error message."
+          endAdornment=<ErrorOutlineIcon color="error" />
```
Comment (swoopertr) [inline frontend/src/components/TextFieldComponents/TextFieldComponents.jsx:45]: What are the colors ? "error" error color?
can you make the selection with enum like; ErrorOutlineIcon.Color.Error

**Review by uparkalau — CHANGES_REQUESTED** [review]

File: frontend/src/components/TextFieldComponents/CustomTextField/CustomTextField.jsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  chips: [],
+  startAdornment: null,
+  endAdornment: null,
+  placeholder: "",
+  labelFontWeight: 600,
+  TextFieldWidth: "100%",
+  inputHeight: "56px",
+};
+
+CustomTextField.propTypes = {
```
[the remark this answers is not in the document]
Comment (thomastepi) [reply frontend/src/components/TextFieldComponents/CustomTextField/CustomTextField.jsx:75]: should i delete the prop types

File: frontend/src/components/TextFieldComponents/CustomTextField/CustomTextField.jsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        }}
+        FormHelperTextProps={{
+          sx: { margin: 0, paddingTop: 1 },
+        }}
+      />
+    </Box>
+  );
+};
+
+CustomTextField.defaultProps = {
```
Comment (erenfn) [inline frontend/src/components/TextFieldComponents/CustomTextField/CustomTextField.jsx:59]: Vlad means this part

File: frontend/src/components/TextFieldComponents/CustomTextField/CustomTextField.jsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { TextField, Chip, Box, InputLabel } from "@mui/material";
+import ChipAdornment from "../Chips/ChipAdornment";
+import "./CustomTextFieldStyles.css";
+
+const CustomTextField = ({
+  labelText,
```
Comment (erenfn) [inline frontend/src/components/TextFieldComponents/CustomTextField/CustomTextField.jsx:8]: Add these defaults here

  labelText: "",
  defaultValue: "",
  helperText: "",
  error: false,
  multiline: false,
  rows: 1,
  chips: [],
  startAdornment: null,
  endAdornment: null,
  placeholder: "",
  labelFontWeight: 600,
  TextFieldWidth: "100%",
  inputHeight: "56px",

**Review by erenfn — CHANGES_REQUESTED** [review]

**Review by uparkalau — APPROVED** [review]

**Review by uparkalau — APPROVED** [review]

**Review by uparkalau — APPROVED** [review]

**Review by erenfn — APPROVED** [review]

**Pull request merged**